libmamba/src/core/run.cpp

```diff
@@ -442,24 +477,38 @@ namespace mamba
             }
 
 #ifndef _WIN32
-            MainExecutor::instance().schedule(
```

Comment:
Why this change?

Reply:
I think I forgot this change from one of the experiments I did. I'll revert it.

Do you know why the signal handler was being set with the `MainExecutor` instead of directly?

Reply:
No 🤷

Reply:
I traced this back to 60d4b3a, but still don't know why the signal was being set in a separate thread.

With the new changes, it seems more difficult to do the modifications correctly in a separate thread. The MainExecutor doesn't seem to have functionality to check if a scheduled task has finished, meaning that it wouldn't be possible to check that modifying and restoring the signal masks happen before the end of the `run_in_environment` call without using some sort of synchronization primitives. At the same time, the signal mask is set on a per-thread basis, so the calls to `pthread_sigmask` would have to anyhow be done on the main thread. What do you think about leaving this as is?

Reply:
@wolfv do you recall why: